pyiron/atomistics/master/elastic.py

```diff
@@ -0,0 +1,385 @@
+    )
+    return strain_higher_terms
+
+def calc_elastic_tensor(
+        strain,
+        stress=None,
+        energy=None,
+        volume=None,
+        rotations=None,
+        return_score=False,
+        max_polynomial_order=None,
+        fit_first_order=False,
+    ):
+    """
+    Calculate 6x6-elastic tensor from the strain and stress or strain and
+    energy+volume.
+
+    Rotations matrices can be added to take box symmetries into account (unit
+    matrix can be added but does not have to be included in the list)
+
+    Args:
+        strain (numpy.ndarray): nx3x3 strain tensors
+        stress (numpy.ndarray): nx3x3 stress tensors
+        energy (numpy.ndarray): n energy values
+        volume (numpy.ndarray): n volume values
+        rotations (numpy.ndarray): mx3x3 rotation matrices
+        return_score (numpy.ndarray): return regression score
+            (cf. sklearn.linear_mode.LinearRegression)
+    """
+    if len(strain)==0:
+        raise ValueError('Not enough points')
+    if rotations is not None:
+        rotations = np.append(np.eye(3), rotations).reshape(-1, 3, 3)
+        _, indices = np.unique(
+            np.round(rotations, 6), axis=0, return_inverse=True
+        )
+        rotations = rotations[np.unique(indices)]
+    else:
+        rotations = np.eye(3).reshape(1, 3, 3)
+    if stress is not None and len(stress)==len(strain):
+        coeff, score = _fit_coeffs_with_stress(
+            strain=strain,
+            stress=stress,
+            rotations=rotations,
+            max_polynomial_order=max_polynomial_order,
+            fit_first_order=fit_first_order,
+        )
+    elif (energy is not None
+          and volume is not None
+          and len(energy)==len(strain)
+          and len(energy)==len(volume)):
+        coeff, score = _fit_coeffs_with_energies(
+            strain=strain,
+            energy=energy,
+            volume=volume,
+            rotations=rotations,
+            max_polynomial_order=max_polynomial_order,
+            fit_first_order=fit_first_order,
+        )
+    else:
+        raise ValueError('Provide either strain and stress, or strain, energy '
+                         + 'and volume of same length.')
+    if return_score:
+        return np.array(coeff)[:,:6], score
+    else:
+        return np.array(coeff)[:,:6]
+
+def get_strain(max_strain=0.05, n_set=10, polynomial_order=3, additional_points=0, normalize=False):
+    strain_lst = np.random.random((n_set, 3, 3))-0.5
+    strain_lst += np.einsum('nij->nji', strain_lst)
+    if normalize:
+        strain_lst = np.einsum('nij,n->nij', strain_lst, 1/np.linalg.norm(strain_lst.reshape(-1, 9), axis=-1))
+    strain_lst *= max_strain
+    m = np.linspace(-1, 1, int(2*polynomial_order+2*additional_points-1))
+    strain_lst = np.einsum('k,nij->nkij', m, strain_lst).reshape(-1, 9)
+    return np.unique(strain_lst, axis=0).reshape(-1, 3, 3)
+
+class ElasticJobGenerator(JobGenerator):
```

Comment:
Nit: IMO, these generator classes (except the ABC `JobGenerator`) should be private.

Reply:
This is part of a broader push I'd like to make to start privatizing stuff that never gets used outside its scope even by the developers (and isn't expected to ever be used outside that scope).

Reply:
Good point